File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/infrastructure/persistence/repositories/JpaSigningKeyRepository.java
package br.com.deltaglobalbank.identity.infrastructure.persistence.repositories;

import java.util.List;
import java.util.UUID;

import br.com.deltaglobalbank.identity.infrastructure.persistence.entities.SigningKeyEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface JpaSigningKeyRepository extends JpaRepository<SigningKeyEntity, UUID> {
    SigningKeyEntity findByKid(String kid);

    List<SigningKeyEntity> findAllByStatus(String status);

    SigningKeyEntity findFirstByStatusOrderByActivatedAtDesc(String status);
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/infrastructure/persistence/repositories/JpaSigningKeyRepository.kt
package br.com.deltaglobalbank.identity.infrastructure.persistence.repositories

import br.com.deltaglobalbank.identity.infrastructure.persistence.entities.SigningKeyEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface JpaSigningKeyRepository : JpaRepository<SigningKeyEntity, UUID> {
    fun findByKid(kid: String): SigningKeyEntity?
    fun findAllByStatus(status: String): List<SigningKeyEntity>
    fun findFirstByStatusOrderByActivatedAtDesc(status: String): SigningKeyEntity?
}
